fix rpm recommends

rbtr opened the pull request:
Fixes goreleaser/goreleaser/issues/1636

`recommends` was not being included in the rpmpack struct so including that field in the nfpm config was ineffective. fixed that so that rpms correctly include the recommended packages.

After applying this patch and rebuilding goreleaser:
```
$ sudo dnf repoquery --recommends ./my-app.rpm
Last metadata expiration check: 2:10:04 ago on Wed 01 Jul 2020 10:28:08 AM EDT.
< the recommended dependency >

$ sudo dnf install ./my-app.rpm               
Last metadata expiration check: 2:10:11 ago on Wed 01 Jul 2020 10:28:08 AM EDT.
Dependencies resolved.
===
[...]
Installing weak dependencies:
 < the recommended dependency >
```

caarlos0 commented:
awesome, thanks for the PR!

do you mind adding tests as well? 

I think we can add recommends [here](https://github.com/goreleaser/nfpm/blob/master/acceptance/testdata/complex.yaml) and an assertion [here](https://github.com/goreleaser/nfpm/blob/master/acceptance/testdata/rpm.complex.dockerfile)...

rbtr commented:
Updated. Since the test package is installed with `rpm`, dependencies are not automatically resolved - it would need to use yum or dnf to automatically install those. The best we can do instead is check that the rpm self-reports its `recommends` correctly. I added that and a similar test for the existing `suggests`.
I unfortunately can't figure out an equivalent way to do this in dpkg, so I have left the .deb test alone.

caarlos0 commented:
awesome, thanks!

caarlos0 commented:
This pull request has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
